src/pages/_app.tsx

```diff
@@ -66,38 +65,29 @@ function ExplorerApp({ Component, ...rest }: ExplorerAppProps): React.ReactEleme
   );
 }
 
-const handleNetworkModeQueryParam = (store: EnhancedStore, appContext: AppContext) => {
-  if (appContext.ctx.pathname === '/_error') return;
+ExplorerApp.getInitialProps = async (appContext: AppContext) => {
+  const appProps = await App.getInitialProps(appContext);
   const query = appContext.ctx.query;
-  const activeNetwork = selectActiveNetwork(store.getState());
-  const networks = selectNetworks(store.getState());
-  console.log('[debug] selectActiveNetworkUrl', selectActiveNetworkUrl(store.getState()));
-  console.log('[debug] activeNetwork', activeNetwork);
-  console.log('[debug] networks', networks);
   const queryNetworkMode = ((Array.isArray(query.chain) ? query.chain[0] : query.chain) ||
     '') as NetworkModes;
-  if (queryNetworkMode !== activeNetwork?.mode || !networks[activeNetwork.url]) {
-    // query param overrides state
-    console.log(
-      '[debug] network conflict',
-      queryNetworkMode,
-      activeNetwork,
-      DEFAULT_NETWORK_MAP,
-      NetworkModeUrlMap
-    );
-    store.dispatch(setActiveNetwork(DEFAULT_NETWORK_MAP[NetworkModeUrlMap[queryNetworkMode]]));
-  }
+  const queryApiUrl = Array.isArray(query.api) ? query.api[0] : query.api;
+  store.dispatch(initialize({ queryNetworkMode, apiUrls: NetworkModeUrlMap, queryApiUrl }));
+  console.log(
```

Comment:
Do we still want to keep all these debug logs? Are they useful?

Reply:
will only keep them until 2.1 is out to help spot issues quickly